UT flake: TestNewUnmanagedBridgeConfigurationResolvesDPUHostRepresentor depends on map iteration order

### Which jobs are flaking?

The unit-tests job. Example failure from a presubmit run on #6712, which only touches an unrelated UDN controller test file:
https://github.com/ovn-kubernetes/ovn-kubernetes/actions/runs/29814063529/job/88581233236?pr=6712

### Which tests are flaking?

`TestNewUnmanagedBridgeConfigurationResolvesDPUHostRepresentor` in `go-controller/pkg/node/bridgeconfig/bridgeconfig_test.go`.

### Since when has it been flaking?

Since the test was introduced by #6555, merged 2026-07-20.

### Reason for failure (if possible)

It's a map iteration order issue in the test's mock expectations, the production code is fine.

The test sets up two mock expectations, one per bridge port:
- `GetRepresentorPortFlavour("eth1")` returns "not a PF representor"
- `GetRepresentorPortFlavour("pfhpf0")` returns `PORT_FLAVOUR_PCI_PF`

The code under test, `SwitchdevDPUOps.GetDPUHostRepInterface`, iterates `portsToInterfaces`, which is a Go map (`map[string][]*vswitchd.Interface` built by `getBridgePortsInterfaces`), and returns as soon as it finds the first PF-flavour interface. Map iteration order is randomized, so whenever `pfhpf0` happens to be visited before `eth1`, the `eth1` mock is never called and mockery's cleanup assertion fails:

```
I0721 08:36:19.348262   28119 dpuops_linux.go:85] DPUOps initialised: Switchdev hardware DPU environment
    SriovnetOps.go:530: FAIL:	GetRepresentorPortFlavour(string)
        		at: [/home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/node/bridgeconfig/bridgeconfig_test.go:140]
    SriovnetOps.go:530: FAIL: 1 out of 2 expectation(s) were met.
        	The code you are testing needs to make 1 more call(s).
--- FAIL: TestNewUnmanagedBridgeConfigurationResolvesDPUHostRepresentor (0.17s)
```

Easy to reproduce locally: `go test -race -run TestNewUnmanagedBridgeConfigurationResolvesDPUHostRepresentor ./pkg/node/bridgeconfig/ -count=10` failed 1 out of 10 runs for me.

### Anything else we need to know?

The fix is to mark the order-dependent `eth1` expectation with `.Maybe()`: whether the uplink gets probed depends on map order, so the test shouldn't require it. The production code doesn't need deterministic iteration, since any PF representor is acceptable.

--- a/go-controller/pkg/node/bridgeconfig/bridgeconfig_test.go
+++ b/go-controller/pkg/node/bridgeconfig/bridgeconfig_test.go
@@ -137,8 +137,11 @@ func TestNewUnmanagedBridgeConfigurationResolvesDPUHostRepresentor(t *testing.T)
 	t.Cleanup(func() {
 		util.SetSriovnetOpsInst(origSriovOps)
 	})
+	// GetDPUHostRepInterface iterates bridge ports in map order and returns on
+	// the first PF representor, so eth1 may or may not be probed before pfhpf0.
 	sriovOps.On("GetRepresentorPortFlavour", "eth1").
-		Return(sriovnet.PortFlavour(sriovnet.PORT_FLAVOUR_UNKNOWN), fmt.Errorf("not a PF representor"))
+		Return(sriovnet.PortFlavour(sriovnet.PORT_FLAVOUR_UNKNOWN), fmt.Errorf("not a PF representor")).
+		Maybe()
 	sriovOps.On("GetRepresentorPortFlavour", "pfhpf0").
 		Return(sriovnet.PortFlavour(sriovnet.PORT_FLAVOUR_PCI_PF), nil)
 

